API/server.ts

```diff
@@ -0,0 +1,186 @@
+  if (requireAuthentication && authHeader !== appconfig['x-auth-token']) return res.sendStatus(403); // Forbidden
+
+  const supportedJudges: Array<string> = [];
+  for (const onlineJudgeRecord in SupportedOnlineJudges) {
+    const judge = SupportedOnlineJudges[onlineJudgeRecord as keyof typeof SupportedOnlineJudges];
+    supportedJudges.push(judge);
+  }
+  res.json({
+    supportedJudges: supportedJudges,
+  });
+  return;
+});
+
+app.post('/submit', async (req, res) => {
+  const authHeader = req.headers['x-auth-token'];
+  if (requireAuthentication && authHeader === undefined) return res.sendStatus(401); //Unauthorized
+  if (requireAuthentication && authHeader !== appconfig['x-auth-token']) return res.sendStatus(403); // Forbidden
+
+  const problemURL: string = req.body.problemURL;
+  const langSolution: ProgrammingLanguage = req.body.langSolution;
+  const solution: string = req.body.solution;
+
+  // Check three parameters are sent
+  if (problemURL === undefined || langSolution === undefined || solution === undefined) {
+    res.json({
+      message: 'Expected three parameters: [problemURL, langSolution, solution]',
+    });
+    return;
+  }
+
+  // Check programming language is supported
+  if (!Object.values(SupportedProgrammingLanguages).includes(langSolution)) {
+    res.json({
+      message: `Programming Language not suported, see /languages`,
+    });
+    return;
+  }
+
+  // Check problem URL contains an online judge keyword
+  let regexChecker: RegExp;
+  let judge: OnlineJudge;
+  if (problemURL.includes(SupportedOnlineJudges.codeforces)) {
+    regexChecker = OnlineJudgesURLRegularExpression.codeforces;
+    judge = SupportedOnlineJudges.codeforces;
+  } else if (problemURL.includes(SupportedOnlineJudges.hackerrank)) {
+    regexChecker = OnlineJudgesURLRegularExpression.hackerrank;
+    judge = SupportedOnlineJudges.hackerrank;
+  } else if (problemURL.includes(SupportedOnlineJudges.kattis)) {
+    regexChecker = OnlineJudgesURLRegularExpression.kattis;
+    judge = SupportedOnlineJudges.kattis;
+  } else {
+    res.json({
+      message: 'Online judge not supported',
+    });
+    return;
+  }
+
+  // Check problem URL matches its corresponding judge pattern
+  if (regexChecker.test(problemURL)) {
+    // Check that the problem exist
+    if (await problemExist(problemURL)) {
+      // Check if the solution is base64 encoded
+      let sol: string = solution;
+      const base64_encoded = 'true' === req.query.base64_encoded?.toString().toLowerCase() ?? 'false';
+      if (base64_encoded) {
+        const b64string = solution;
+        sol = Buffer.from(b64string, 'base64').toString('utf-8');
+      }
+
+      // Trying to get an available account
+      let username: string;
+      let password: string;
+      let account: AccountAvailable;
+
+      try {
+        account = await getAccountAvailable(judge);
+        let user = appconfig.judgeAccounts[account.userID];
```

Comment:
What happens with all the variables in the configuration if they're not set? You can check the config file at the beginning and throw the errors before start the api and wait the error while executing.